src/API/Authentication.php

```diff
@@ -150,198 +159,199 @@ public function email_passwordless_start($email, $type, $authParams = [])
+        }
+
+        if (!isset($options['client_secret'])) {
+            $options['client_secret'] = $this->clientSecret;
+        }
+
+        if (!isset($options['grant_type'])) {
+            throw new ApiException('grant_type is mandatory');
+        }
+
+        $response = $this->httpClient->post('/oauth/token', [], json_encode($options));
+
+        return ResponseMediator::getContent($response);
+    }
+
+    /**
+     * Makes a call to the `oauth/token` endpoint with `authorization_code` grant type.
+     *
+     * @param $code
+     * @param $redirectUri
+     *
+     * @return array|string
+     */
+    public function code_exchange($code, $redirectUri)
+    {
+        $options = [];
+
+        $options['client_secret'] = $this->clientSecret;
+        $options['redirect_uri'] = $redirectUri;
+        $options['code'] = $code;
+        $options['grant_type'] = 'authorization_code';
+
+        return $this->oauth_token($options);
+    }
+
+    /**
+     * Makes a call to the `oauth/token` endpoint with `password-realm` grant type.
+     *
+     * @method login
+     *
+     * @param {Object} options:
+     * @param {Object} options.username
+     * @param {Object} options.password
+     * @param {Object} options.realm
+     * @param {Object} options.scope [optional]
+     * @param {Object} options.audience [optional]
+     */
+    public function login($options)
+    {
+        if (!isset($options['username'])) {
+            throw new ApiException('username is mandatory');
+        }
+
+        if (!isset($options['password'])) {
+            throw new ApiException('password is mandatory');
+        }
+
+        if (!isset($options['realm'])) {
+            throw new ApiException('realm is mandatory');
+        }
+
+        $options['grant_type'] = 'http://auth0.com/oauth/grant-type/password-realm';
+
+        return $this->oauth_token($options);
+    }
+
+    /**
+     * Makes a call to the `oauth/token` endpoint with `password` grant type.
+     *
+     * @method login_with_default_directory
+     *
+     * @param {Object} options: https://auth0.com/docs/api-auth/grant/password
+     * @param {Object} options.username
+     * @param {Object} options.password
+     * @param {Object} options.scope [optional]
+     * @param {Object} options.scope [audience]
+     */
+    public function login_with_default_directory($options)
+    {
+        if (!isset($options['username'])) {
+            throw new ApiException('username is mandatory');
+        }
+
+        if (!isset($options['password'])) {
+            throw new ApiException('password is mandatory');
+        }
+
+        $options['grant_type'] = 'password';
+
+        return $this->oauth_token($options);
+    }
+
+    /**
+     * Makes a call to the `oauth/token` endpoint with `client_credentials` grant type.
+     *
+     * @method client_credentials
+     *
+     * @param {Object} options: https://auth0.com/docs/api-auth/grant/client-credentials
```

Comment:
I agree. I only changed indentation on this line. I'll improve this class in a next PR.